pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,48 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec         `json:"spec,omitempty"`
+	Status            TrainedModelDeployStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+	// Explainer model spec
+	ExplainerModel ModelSpec `json:"explainerModel,omitempty"`
+}
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Default to latest
+	ModelVersionPolicy ModelVersionPolicy `json:"modelVersionPolicy,omitempty"`
+	// ML framework name
+	// +kubebuilder:webhooks:Enum={"tensorflow","pytorch","sklearn","onnx","xgboost", "custom"}
+	Framework string `json:"framework"`
+	// Framework version for the trained model
+	FrameworkVersion string `json:"frameworkVersion,omitempty"`
+	// Defaults to requests and limits of 1CPU, 2Gb MEM.
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+}
+type ModelVersionPolicy struct {
+	// +kubebuilder:webhooks:Enum={"all","latest","specific"}
+	ModelVersionPolicy string `json:"modelVersionPolicy"`
+	// A list of specific model versions to serve, it is required when model version policy is "specific"
+	ModelVersions []string `json:"modelVersions,omitempty"`
+}
+type TrainedModelDeployStatus struct {
```

Comment:
Consider breaking this out into its own definition where we can implement something like propagate status. Also, consider "TrainedModelStatus"

Reply:
Sounds good.